src/main.cc

```diff
@@ -584,7 +585,26 @@ int main( int argc, char ** argv )
 
   MainWindow m( cfg );
 
-  app.addDataCommiter( m );
+  /// Session manager things.
+  // Redirect commit data request to Mainwindow's handler.
+  QObject::connect(
+    &app,
+    &QGuiApplication::commitDataRequest,
+    &m,
+    [ &m ]( QSessionManager & ) {
+      m.commitData();
+    },
+    Qt::DirectConnection );
```

Comment:
any difference between DirectConnection and default?

Reply:
I don't think so, `app` and `m` lives in the same thread, so the default would be `DirectionConnection` anyway.

But, the doc have a big note there, so I didn't delete it :)

https://doc.qt.io/qt-6/qguiapplication.html#commitDataRequest